exp/internal/controllers/machinepool_controller_phases_test.go

```diff
@@ -1448,7 +1448,7 @@ func TestReconcileMachinePoolMachines(t *testing.T) {
 				clusterv1.ClusterNameLabel:     clusterName,
 				clusterv1.MachinePoolNameLabel: machinePool.Name,
 			}
-			g.Expect(env.GetAPIReader().List(ctx, machineList, client.InNamespace(cluster.Namespace), client.MatchingLabels(labels))).To(Succeed())
```

Comment:
or am i misunderstanding this, does `env.GetAPIReader().List` not hit the cache at all?

Reply:
that's correct. API reader skips the cache and goes direct to API server.

Reply:
Not sure I follow the reasoning why using a cached client now would make the tests more stable? (or maybe @cahillsf you understood it the other way around? :))